### PR TITLE
Some improvements to LayoutFormatterPreferences

### PR DESCRIPTION
**Pull request by oscargus**
Some suggestions from by @koppor in #1695 which I didn't fully understand then. Also final Globals call in `logic`removed.

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/layout/LayoutFormatterPreferences.java [hunk 1]
```diff
@@ -45,15 +45,23 @@ public JournalAbbreviationPreferences getJournalAbbreviationPreferences() {
...
+
+    public void clearCustomExportNameFormatters() {
+        customExportNameFormatters.clear();
+    }
+
+    public void putCustomExportNameFormatter(String formatterName, String contents) {
+        customExportNameFormatters.put(formatterName, contents);
+    }
+
+    public String getCustomExportNameFormatter(String className) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/LayoutFormatterPreferences.java]: Should this parameter be called `formatterName`?

File: src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java [hunk 2]
```diff
@@ -585,14 +585,13 @@ private LayoutFormatter getLayoutFormatterByName(String name) throws Exception {
...
-                String contents = prefs.getCustomExportNameFormatters().get(className);
-                if (contents != null) {
-                    NameFormatter nf = new NameFormatter();
-                    nf.setParameter(contents);
-                    results.add(nf);
-                    continue;
-                }
+
+            String contents = prefs.getCustomExportNameFormatter(className);
+            if (contents != null) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java]: How much effort would it be to switch to `Optional` at `prefs.getCustomExportNameFormatter`?

Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java]: Not much. However, the `continue` prohibits the use of `ifPresent` so there will still be a `if (contents.isPresent())`. Still that is probably better.

File: src/main/java/net/sf/jabref/logic/layout/LayoutFormatterPreferences.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/layout/LayoutFormatterPreferences.java]: Yes. Will change in LayoutEntry as well (I just used the name from refactoring).

**Comment by oscargus** [comment]
Fixed the comments.

**Pull request merged**